### PR TITLE
refactor: Replace PDF course certificate view code

### PR DESCRIPTION
**Pull request by ruzniaievdm**
[JIRA](https://openedx.atlassian.net/browse/DEPR-157)
[YT](https://youtrack.raccoongang.com/issue/OeX_Depr-12)

Removed fields `download_url` and `download_uuid` from model `GeneratedCertificate`
This request does not include changes below, need to discuss this points:
1. ExampleCertificate, this model also has `download_url` field [link to pr where part functionality is removed](https://github.com/openedx/edx-platform/pull/28332)
2. `downloadable` status from `CertificateStatuses` need be removed or renamed, due to logic mismatch, since in this request we will remove the ability to download a certificate

### CODE REVIEW

File: lms/djangoapps/certificates/utils.py [hunk 1]
```diff
@@ -57,7 +57,7 @@ def emit_certificate_event(event_name, user, course_id, course_overview=None, ev
     data = {
         'user_id': user.id,
         'course_id': str(course_id),
-        'certificate_url': get_certificate_url(user.id, course_id, uuid=event_data['certificate_id'])
+        'certificate_url': get_certificate_url(course_id, uuid=event_data['certificate_id'])
```
Comment (idegtiarov) [inline lms/djangoapps/certificates/utils.py:60]: I'm curious why we still need `get_certificate_url` function, shouldn't it be removed as well?

Comment (ruzniaievdm) [reply lms/djangoapps/certificates/utils.py:60]: `get_certificate_url` function used to get based certificate URL for HTML representation of certificate

File: lms/djangoapps/certificates/apis/v0/tests/test_views.py [hunk 2]
```diff
@@ -393,10 +383,9 @@ def test_with_no_certificate_configuration(self):
             requesting_user=self.global_staff,
             requested_user=self.student,
         )
-        kwargs = {"certificate_uuid": self.cert.verify_uuid}
-        expected_download_url = reverse('certificates:render_cert_by_uuid', kwargs=kwargs)
-        self.assert_success_response_for_student(response, download_url=expected_download_url)
+        self.assert_success_response_for_student(response)
 
+    #TODO
```
[the remark this answers is not in the document]
Comment (ruzniaievdm) [reply lms/djangoapps/certificates/apis/v0/tests/test_views.py:388]: my bad, need to remove

File: common/djangoapps/student/tests/test_certificates.py [hunk 3]
```diff
@@ -225,41 +174,12 @@ def test_post_to_linkedin_visibility(self):
...
-        self._check_can_not_download_certificate()
-
-
 @ddt.ddt
 @unittest.skipUnless(settings.ROOT_URLCONF == 'lms.urls', 'Test only valid in lms')
 class CertificateDisplayTestLinkedHtmlView(CertificateDisplayTestBase):
     """
     Tests of linked student certificates.
     """
-
```
[the remark this answers is not in the document]
Comment (ruzniaievdm) [reply common/djangoapps/student/tests/test_certificates.py:262]: You are right, I will change

File: lms/djangoapps/mobile_api/users/serializers.py [hunk 4]
```diff
@@ -103,9 +103,7 @@ def get_certificate(self, model):
         certificate_info = certificate_downloadable_status(model.user, model.course_id)
         if certificate_info['is_downloadable']:
             return {
-                'url': self.context['request'].build_absolute_uri(
-                    certificate_info['download_url']
-                ),
+                'url': '',
```
Comment (dyudyunov) [inline lms/djangoapps/mobile_api/users/serializers.py:106]: Interesting)
This field method looks pretty useless, isn't it?) I mean it returns {} or {url: ''}

I'm curious if we could get rid of it

File: lms/djangoapps/mobile_api/users/tests.py [hunk 5]
```diff
@@ -370,37 +348,6 @@ def test_no_certificate(self):
...
-
-    @patch.dict(settings.FEATURES, {'CERTIFICATES_HTML_VIEW': True, 'ENABLE_MKTG_SITE': True})
-    def test_pdf_certificate_with_html_cert_enabled(self):
-        """
-        Tests PDF certificates with CERTIFICATES_HTML_VIEW set to True.
-        """
-        self.verify_pdf_certificate()
-
-    @patch.dict(settings.FEATURES, {'CERTIFICATES_HTML_VIEW': True, 'ENABLE_MKTG_SITE': True})
-    def test_web_certificate(self):
```
Comment (dyudyunov) [inline lms/djangoapps/mobile_api/users/tests.py:388]: Why do you deleting this test? Looks like it cheks the certificate web view URL, not the download URL

**Review by dyudyunov — APPROVED** [review]
I wonder if you check the certificate generation process - if there were any tasks for pdf certificate generation? Or it was already removed before? I'm asking because I didn't see this functionality deletion in the PR.

And AFAIK certificate URLs should be available on the profile page (for web certificates). Did you check if they are still available? I'm asking because I saw you deleted certificate URLs block from the profile and just want to doublecheck

File: lms/djangoapps/certificates/tests/test_api.py [hunk 6]
```diff
@@ -472,23 +420,14 @@ def test_get_certificates_for_user(self):
         Test to get all the certificates for a user
         """
         certs = get_certificates_for_user(self.student.username)
```
[the remark this answers is not in the document]
Comment (ruzniaievdm) [reply lms/djangoapps/certificates/tests/test_api.py:422]: Yes, it was possible to do so, before that were two certificates to use, so I left the access by index

File: lms/djangoapps/mobile_api/users/tests.py (hunk 5 above)
Comment (ruzniaievdm) [reply lms/djangoapps/mobile_api/users/tests.py:388]: This is the test where i left empty url in `serializer.py`

**Pull request closed without merging**